rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/security/ToggleSecurityFilter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.coordinator.security;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.ext.Provider;
+import java.security.Principal;
+
+@PreMatching
+@Provider
+public class ToggleSecurityFilter implements ContainerRequestFilter {
+
+    @Inject
+    @ConfigProperty(name = "lra.security.enabled", defaultValue = "false")
+    Instance<Boolean> isSecurityEnabled;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        if (!isSecurityEnabled.get()) {
+            SecurityContext securityContext = new SecurityContext() {
+                @Override
+                public Principal getUserPrincipal() {
+                    return new Principal() {
+                        @Override
+                        public String getName() {
+                            return "no security";
+                        }
+                    };
+                }
+
+                @Override
+                public boolean isUserInRole(String role) {
+                    return true;
```

Comment:
What could be making use of this SecurityContext and subsequently calling this method?

Reply:
Actually here the `if` block is used to bypass the security, this means that user has disabled security.